domain/credential/state/state.go

```diff
@@ -373,7 +354,11 @@ AND    cloud_credential.cloud_uuid = (
 
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
 		var outcome sqlair.Outcome
-		terms := credentialKeyMap(key)
+		terms := sqlair.M{
```

Comment:
Do we want to remove the `sqlair.M` as part of this PR ?

Reply:
There's more work to do and it's beyond the scope of the PR. ie there's here and other sql.M uses and for this case the logic is not a straight swap so I left it as it for now.